tools/FindPipelinesUsingRetiredImages/QueryJobHistoryForRetiredImages.ps1

```diff
@@ -0,0 +1,102 @@
+param (
+    $accountUrl,
+    $pat
+)
+
+# Create the VSTS auth header
+$base64authinfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":$pat"))
+$vstsAuthHeader = @{"Authorization"="Basic $base64authinfo"}
+$allHeaders = $vstsAuthHeader + @{"Content-Type"="application/json"; "Accept"="application/json"}
+
+try
+{
+	$result = Invoke-WebRequest -Headers $allHeaders -Method GET "$accountUrl/_apis/DistributedTask/pools?api-version=5.0-preview"
+	if ($result.StatusCode -ne 200)
+    {
+		echo $result.Content
+		throw "Failed to query pools"
+	}
+	$resultJson = ConvertFrom-Json $result.Content
+    $azurePipelinesPoolId = 0
+    foreach($pool in $resultJson.value)
+    {
+        if ($pool.name -eq "Azure Pipelines")
+        {
+            $azurePipelinesPoolId = $pool.id
+            break
+        }
+    }
+
+    if ($azurePipelinesPoolId -eq 0)
+    {
+        throw "Failed to find Azure Pipelines pool"
+    }
+    
+    Write-Host ("Azure Pipelines Pool Id: " + $azurePipelinesPoolId)
+
+    $msg = 'Query next 200 jobs? (y/n)'
+    $response = 'y'
+    $continuationToken = 0
```

Comment:
Do we want to make this a parameter to the script and print it out later (or maybe just on failure)? That way if the script fails in the middle for some reason the user can pick back up where it started

Reply:
Good idea.